builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -1276,3 +1281,26 @@ func expandAzureRmVirtualMachineOsDisk(d *schema.ResourceData) (*compute.OSDisk,
 
 	return osDisk, nil
 }
+
+func findStorageAccountResourceGroup(meta interface{}, storageAccountName string) (string, error) {
+	client := meta.(*ArmClient).resourceFindClient
+	filter := fmt.Sprintf("name eq '%s' and resourceType eq 'Microsoft.Storage/storageAccounts'", storageAccountName)
+	expand := ""
+	var pager *int32
+	rf, err := client.List(filter, expand, pager)
+
+	if err != nil {
+		return "", fmt.Errorf("Error making resource request for query %s: %s", filter, err)
+	}
+
+	results := *rf.Value
+	if len(results) != 1 {
+		return "", fmt.Errorf("Wrong number of results making resource request for query %s:  %s", filter, len(results))
+	}
+
+	// Storage Account ID is in the form
+	//   /subscriptions/[SUBSCRIPTION_ID]/resourceGroups/[RESOURCE_GROUP]/providers/Microsoft.Storage/storageAccounts/[STORAGE_ACCOUNT]
+	idSplit := strings.Split(strings.TrimPrefix(*results[0].ID, "/"), "/")
+	storageAccountResourceGroupName := idSplit[3]
```

Comment:
Using `parseAzureResourceID` would be cleaner here, would allow you to catch any errors too.